src/chains/celo/utils.ts

```diff
@@ -43,7 +43,9 @@ export function isCIP42(
   transaction: CeloTransactionSerializable | CeloTransactionRequest,
 ): transaction is TransactionSerializableCIP42 {
   // Enable end-user to force the tx to be considered as a cip42
-  if (transaction.type) return transaction.type === 'cip42'
+  if (transaction.type === 'cip42') {
```

Comment:
maybe add a comment explaining why this is necessary and is written this way to prevent future modifications